prompt-service/src/unstract/prompt_service/main.py

```diff
@@ -641,30 +641,48 @@ def prompt_processor() -> Any:
     return structured_output
 
 
-def simple_retriver(  # type:ignore
+def simple_or_subq_retriever(  # type:ignore
     output: dict[str, Any],
     doc_id: str,
     llm_helper: ToolLLM,
     llm_li: Optional[LLM],
     vector_index,
+    retrival_type: str,
 ) -> tuple[str, str]:
     prompt = construct_prompt_for_engine(
         preamble=output["preamble"],
         prompt=output["promptx"],
         postamble=output["postamble"],
         grammar_list=output["grammar"],
     )
-    subq_prompt = (
-        f"Generate a sub-question from the following verbose prompt that will"
-        f" help extract relevant documents from a vector store:\n\n{prompt}"
-    )
+    context_prompt = ""
+    if retrival_type is PSKeys.SIMPLE:
+        context_prompt = prompt
+    if retrival_type is PSKeys.SUBQUESTION:
+        context_prompt = (
+            f"Generate a sub-question from the following verbose prompt that will"
+            f" help extract relevant documents from a vector store:\n\n{prompt}"
+        )
```

Comment:
@harini-venkataraman We need to invoke LLM via `run_completion()` only for subquestion retriever. For the simple retriever, we proceed with original prompt as is.

```suggestion
    if retrival_type is PSKeys.SUBQUESTION:
        subq_prompt = (
            f"Generate a sub-question from the following verbose prompt that will"
            f" help extract relevant documents from a vector store:\n\n{prompt}"
        )
        prompt = run_completion(
            llm_helper=llm_helper,
            llm_li=llm_li,
            prompt=context_prompt,
            adapter_instance_id=output[PSKeys.LLM],
        )

    context = _retrieve_context(output, doc_id, vector_index, prompt)

    answer = construct_and_run_prompt(  # type:ignore
        output,
        llm_helper,
        llm_li,
        text,
        "promptx",
    )

    return (answer, context)
```